bare_metal_tests/TESTS/bringup/uart/test.py

```diff
@@ -0,0 +1,330 @@
+"""
+mbed SDK
+Copyright (c) 2017 ARM Limited
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import time
+import serial
+import sys
+import mbed_lstools
+from subprocess import Popen, PIPE, STDOUT
+import shutil
+import argparse
+import ntpath
+import json
+import os
+import logging
+
+"""! UART TEST FOR GREEN TEA
+
+  TEST SCENARIO
+  This part of the test is executed on the host (PC).
+ 
+  After startup it will try to communicate via UART with the mbed device.
+  This part of the test waits for the expected message and responds with
+  the same message. 
+  4 test messages should be received. Last message ("PASSED") should inform
+  that on the mbed side also all messages have been received.
+ 
+  To perform this test with serial port baud rate different than default (9600),
+  call host test with appropriate --baudrate parameter.
+  
+  Example usage:
+      test.py -m K64F -t GCC_ARM -b 9600
+  This command will run test on K64F platfom using GCC_ARM toolchain and
+  serial connection speed equal to 9600 b/s.
+  
+  For more options:
+      test.py -h
+      
+"""
+
+serial_port = None
+RETRY_LIMIT = 10
+
+# List of expected messages from the mbed device.
+message_list = ['x\0', 
+                'mbed\0',
+                'Green Tea UART bear metal test.\0', 
+                'Long Message-Long Message-Long Message-Long Message-Long Message-Long Message-Long Message-Long Message-Long Message--.\0',
+                'PASSED\0']
+
+def exit():
+    if(serial_port != None and serial_port.isOpen() == True):
+        serial_port.close()
+    sys.exit()
+        
+def create_test_header_file(baudrate):
+    """! Create header file with the specified baud rate.
+    
+    @param baudrate specified transmission baud rate.
+    """
+    file = open('test_config.h','w') 
+    file.write('/* This file has been automatically generated by the python script. */\n')
+    file.write('/* This file contains serial port baud rate. */\n')
+    file.write('#define BAUD_RATE %s\n' % baudrate)
```

Comment:
This could be done by `mbed compile ... -DBAUD_RATE=%s`.

Reply:
That is much better. Fixed as suggested.